test/models/iterator_test.js

```diff
@@ -0,0 +1,35 @@
+const { expect } = require('chai');
+const fs = require('fs');
+const path = require('path');
+
+const AsyncAPIDocument = require('../../lib/models/asyncapi');
+
+describe('AsyncAPIDocument', function () {
+  describe('constructor', function() {
+    it('Should traverse patternProperties, definitions, dependencies ', function () {
+      const doc = JSON.parse(fs.readFileSync(path.resolve(__dirname, '../good/iterator.json'),'utf8'));
+      const d = new AsyncAPIDocument(doc);
```

Comment:
Why sonar cloud is giving me a bug here stating `Replace AsyncAPIDocument with a constructor function.`

Reply:
I actually have no idea, to me, this sounds like a false positive.

@derberg any idea 🤔

Reply:
Yeah! the same bug is reported 91 times in every file by sonar in this PR too: [integrate eslint jsdoc validation](https://github.com/asyncapi/parser-js/pull/305)

For Reference:

<img width="1050" alt="Screenshot 2021-05-25 at 9 16 53 PM" src="https://user-images.githubusercontent.com/53009722/119528094-8a6fd780-bd9e-11eb-93c9-8467b78bcdd4.png">

Reply:
did you click `Why is this an issue?` link? the answer is there 😛 
although confusing is that actually `AsyncAPIDocument` has a constructor that has `@constructor` jsdoc. Maybe the problem here is that `AsyncAPIDocument` extends `Base` that doesn't have a constructor with `@constructor` jsdoc. Try this path.

Reply:
was figuring a way to fix that but couldn't, thanks for pointing the workaround and now it's done 🙂